components/GlobalButton/GlobalButton.styled.js

```diff
@@ -0,0 +1,49 @@
+import styled from "styled-components";
+
+export const globalButton = styled.button`
+  background-color: ${(props) => {
+    switch (props.type) {
+      case "submit":
+        return `var(--button-submit)`;
+      case "edit":
+        return `var(--button-edit)`;
+      case "succes":
+        return `var(--button-save)`;
+      case "cancel":
+        return `var(--button-cancel)`;
+      case "danger":
+        return `var(--button-delete)`;
+      default:
+        return `var(--button-background)`;
+    }
+  }};
+  color: ${(props) => (props.textColor === "white" ? "white" : "black")};
+
+  padding: 10px 20px;
+  border-radius: 8px;
+  box-shadow: 0 3px 4px rgba(0, 0, 0, 0.3);
+  border: none;
+  min-height: 48px;
+  min-width: 48px;
+  cursor: pointer;
+  font: 600 18px var(--font-family);
+
+  &:hover {
+    background-color: ${(props) => {
+      switch (props.type) {
+        case "submit":
+          return `var(--button-submit)`;
+        case "edit":
+          return `var(--button-edit-hover)`;
+        case "success":
+          return `var(--button-save-hover)`;
+        case "cancel":
+          return `var(--button-cancel-hover)`;
+        case "danger":
+          return `var(--button-delete-hover)`;
+        default:
+          return `var(--button-hover)`;
+      }
+    }};
+  }
```

Comment:
you can combine both switch statements which would drastically improve the readability of the code:

example: 
```js

${props => {
  switch(props.type) {
    case: "submit":
      return css`
         background-color: var(--button-submit);
         &:hover {
           background-color: var(--button-submit-hover);
         }
      `
      case "edit":
        return ...
  }
}

Reply:
as you might have seen, the switch statement crates a deeply nested structure. you can prevent this by using a different method:

```js 
${props => props.type === 'submit' && css`
   background-color: var(--button-submit);
   &:hover {
     background-color: var(--button-submit-hover);
   }
`}

${props => props.type === 'edit' && css`
   background-color: var(--button-edit);
   &:hover {
     background-color: var(--button-edit-hover);
   }
`}
```
This has only 1 instead of 4 levels of indentation, which might improve the readability. You can of course decide which path to take here.